tests/beta/test_default_qubit_tf.py

```diff
@@ -765,51 +782,8 @@ def test_hermitian(self, theta, phi, varphi, tol):
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
 
-class TestProbability:
-    """Unit tests for device probability."""
-
-    marginal_test_data = [
-        (tf.constant([0.1, 0.2, 0.3, 0.4]), tf.constant([0.4, 0.6]), [1]),
-        (tf.constant([0.1, 0.2, 0.3, 0.4]), tf.constant([0.3, 0.7]), [0]),
-        (
-            tf.constant(
-                [
-                    0.17794671,
-                    0.06184147,
-                    0.21909549,
-                    0.04932204,
-                    0.19595214,
-                    0.19176834,
-                    0.08495311,
-                    0.0191207,
-                ]
-            ),
-            tf.constant([0.3970422, 0.28090525, 0.11116351, 0.21088904]),
-            [2, 0],
-        ),
-    ]
-
-    def test_correct_marginal_no_wires(self, init_state, tol):
-        """Test that passing wires=None to the marginal probability method
-        simply returns the original probability."""
-        dev = DefaultQubitTF(wires=2)
-        dev._state = tf.reshape(init_state(2), [2] * 2)
-        res = dev.marginal_prob(tf.abs(dev._state) ** 2, wires=None)
-        expected = tf.abs(dev._state) ** 2
-        assert np.allclose(res, expected, atol=tol, rtol=0)
-
-    @pytest.mark.parametrize("probs, marginals, wires", marginal_test_data)
-    def test_correct_marginal_prob(self, probs, marginals, wires, tol):
-        """Test that the marginal probability method returns expected
-        results."""
-        num_wires = int(np.log2(len(probs)))
-        dev = DefaultQubitTF(wires=num_wires)
-        res = dev.marginal_prob(tf.reshape(probs, [2] * num_wires), wires=wires)
-        assert np.allclose(res, marginals, atol=tol, rtol=0)
-
-

```

Comment:
removed because these methods are no longer implemented? I suppose the cross-plugin integration tests will be the true place where this kind of stuff will live?

Reply:
Yep. This code is now only present in `DefaultQubit`, and is already being unit-tested over there.

> I suppose the cross-plugin integration tests will be the true place where this kind of stuff will live?

I feel like there are two types of tests for the plugins:

* Unit tests/device-specific integration tests, that call the device-level units directly. These should live near the code they are testing, and will have to be bespoke for each plugin (e.g., we can't have the cross-plugin integration tests calling and checking `device.apply()`, `device.marginal_prob()`, as there are often slight differences in behaviour and implementation).

* Functional tests, that check for the following:

  - Device loading and QNode binding
  - Gate application
  - Expectations, variances, samples, and probabilities
  - Tensor observables
  - Numeric observables
  - PassthruQNode integration (where relevant)

  These tests should all be written using QNodes, ensuring full compatibility with all devices, and avoiding differences in implementation. This will also be more maintainable --- if we wish to change the device API in the future, or have different device classes, the tests will not need to be modified.

Reply:
A majority of the tests in the test suite fall into the 'functional tests' category, so will end up there. We have _very little actual_ unit testing of default qubit.